src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -0,0 +1,463 @@
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        use self::Error::*;
+
+        match self {
+            CpuId(e) => write!(f, "Cpuid error: {:?}", e),
+            LocalIntConfiguration(e) => write!(
+                f,
+                "Cannot set the local interruption due to bad configuration: {:?}",
+                e
+            ),
+            VcpuFd(e) => write!(f, "Cannot open the VCPU file descriptor: {}", e),
+            MSRSConfiguration(e) => write!(f, "Error configuring the MSR registers: {:?}", e),
+            REGSConfiguration(e) => write!(
+                f,
+                "Error configuring the general purpose registers: {:?}",
+                e
+            ),
+            SREGSConfiguration(e) => write!(f, "Error configuring the special registers: {:?}", e),
+            FPUConfiguration(e) => write!(
+                f,
+                "Error configuring the floating point related registers: {:?}",
+                e
+            ),
+            VcpuGetDebugRegs(e) => write!(f, "Failed to get KVM vcpu debug regs: {}", e),
+            VcpuGetLapic(e) => write!(f, "Failed to get KVM vcpu lapic: {}", e),
+            VcpuGetMpState(e) => write!(f, "Failed to get KVM vcpu mp state: {}", e),
+            VcpuGetMsrs(e) => write!(f, "Failed to get KVM vcpu msrs: {}", e),
+            VcpuGetRegs(e) => write!(f, "Failed to get KVM vcpu regs: {}", e),
+            VcpuGetSregs(e) => write!(f, "Failed to get KVM vcpu sregs: {}", e),
+            VcpuGetVcpuEvents(e) => write!(f, "Failed to get KVM vcpu event: {}", e),
+            VcpuGetXcrs(e) => write!(f, "Failed to get KVM vcpu xcrs: {}", e),
+            VcpuGetXsave(e) => write!(f, "Failed to get KVM vcpu xsave: {}", e),
+            VcpuGetCpuid(e) => write!(f, "Failed to get KVM vcpu cpuid: {}", e),
+            VcpuSetCpuid(e) => write!(f, "Failed to set KVM vcpu cpuid: {}", e),
+            VcpuSetDebugRegs(e) => write!(f, "Failed to set KVM vcpu debug regs: {}", e),
+            VcpuSetLapic(e) => write!(f, "Failed to set KVM vcpu lapic: {}", e),
+            VcpuSetMpState(e) => write!(f, "Failed to set KVM vcpu mp state: {}", e),
+            VcpuSetMsrs(e) => write!(f, "Failed to set KVM vcpu msrs: {}", e),
+            VcpuSetRegs(e) => write!(f, "Failed to set KVM vcpu regs: {}", e),
+            VcpuSetSregs(e) => write!(f, "Failed to set KVM vcpu sregs: {}", e),
+            VcpuSetVcpuEvents(e) => write!(f, "Failed to set KVM vcpu event: {}", e),
+            VcpuSetXcrs(e) => write!(f, "Failed to set KVM vcpu xcrs: {}", e),
+            VcpuSetXsave(e) => write!(f, "Failed to set KVM vcpu xsave: {}", e),
+        }
+    }
+}
+
+type Result<T> = result::Result<T, Error>;
+
+/// A wrapper around creating and using a kvm-based VCPU.
+pub struct KvmVcpu {
+    pub index: u8,
+    pub fd: VcpuFd,
+
+    pub pio_bus: Option<devices::Bus>,
+    pub mmio_bus: Option<devices::Bus>,
+
+    msr_list: MsrList,
+    supported_cpuid: CpuId,
```

Comment:
nit: this is only used to configure for boot, the `configure` function is also arch specific so it could simply take this as a param to be used locally. That would avoid unnecessarily carrying this around in memory.